src/utils/CameraUtils.js

```diff
@@ -0,0 +1,340 @@
+import * as THREE from 'three';
+import TWEEN from '@tweenjs/tween.js';
+import DEMUtils from './DEMUtils';
+import { MAIN_LOOP_EVENTS } from '../Core/MainLoop';
+import Coordinates from '../Core/Geographic/Coordinates';
+
+THREE.Object3D.DefaultUp = new THREE.Vector3(0, 0, 1);
+const targetPosition = new THREE.Vector3();
+const rigs = [];
+const slerp = [];
+
+const deferred = () => {
+    let resolve;
+    return { promise: new Promise((r) => { resolve = r; }), resolve };
+};
+
+// Wrap angle in degrees to [-180 180]
+function wrapTo180(angle) {
+    return angle - Math.floor((angle + 179.0) / 360) * 360;
+}
+
+function tileLayer(view) {
+    return view.getLayers(l => l.protocol == 'tile')[0];
+}
+
+// fix coord.altitude < 0 and !pickedPosition (with plane and ellipsoid intersection)
+function getGroundTargetFromCamera(view, camera, target) {
+    camera.updateMatrixWorld(true);
+    const pickedPosition = view.getPickingPositionFromDepth();
+    const range = pickedPosition && !isNaN(pickedPosition.x) ? camera.position.distanceTo(pickedPosition) : 100;
+    camera.localToWorld(target.set(0, 0, -range));
+}
+
+function getRig(view, camera) {
+    rigs[camera.uuid] = rigs[camera.uuid] || new RigCamera(view, camera);
+    return rigs[camera.uuid];
+}
+
+function proxyProperty(view, camera, rig, key) {
+    rig.proxy.position[key] = camera.position[key];
+    Object.defineProperty(camera.position, key, {
+        get: () => rig.proxy.position[key],
+        set: (newValue) => {
+            rig.removeProxy(view, camera);
+            camera.position[key] = newValue;
+        },
+    });
+}
+
+function RigCamera() {
```

Comment:
This needs a thorougher description.

Reply:
... and a rename as it is not a camera : `CameraRig` ?